pkg/client/daemon/dns/dns.go

```diff
@@ -22,23 +28,103 @@ type Server struct {
 	fallback     *dns.Conn
 	resolve      Resolver
 	requestCount int64
+	cache        *sync.Map
+	recursive    int32 // 0 = never tested, 1 = not recursive, 2 = recursive
```

Comment:
I'd rather not do that because a) it's very internal, and b) then I'll also need to do type casting on all the atomic load and store operations which makes the code less readable.